// @aws-sdk/client-lex-runtime-service/dist-es/models/models_0.js
import { SENSITIVE_STRING, } from "@smithy/smithy-client";
import { LexRuntimeServiceServiceException as __BaseException } from "./LexRuntimeServiceServiceException";
export class BadRequestException extends __BaseException {
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "BadRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, BadRequestException.prototype);
    }
}
export class ConflictException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
export class InternalFailureException extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalFailureException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalFailureException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalFailureException.prototype);
    }
}
export class LimitExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
export class NotFoundException extends __BaseException {
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "NotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
export const FulfillmentState = {
    FAILED: "Failed",
    FULFILLED: "Fulfilled",
    READY_FOR_FULFILLMENT: "ReadyForFulfillment",
};
export const MessageFormatType = {
    COMPOSITE: "Composite",
    CUSTOM_PAYLOAD: "CustomPayload",
    PLAIN_TEXT: "PlainText",
    SSML: "SSML",
};
export const DialogActionType = {
    CLOSE: "Close",
    CONFIRM_INTENT: "ConfirmIntent",
    DELEGATE: "Delegate",
    ELICIT_INTENT: "ElicitIntent",
    ELICIT_SLOT: "ElicitSlot",
};
export const ConfirmationStatus = {
    CONFIRMED: "Confirmed",
    DENIED: "Denied",
    NONE: "None",
};
export class BadGatewayException extends __BaseException {
    constructor(opts) {
        super({
            name: "BadGatewayException",
            $fault: "server",
            ...opts,
        });
        this.name = "BadGatewayException";
        this.$fault = "server";
        Object.setPrototypeOf(this, BadGatewayException.prototype);
        this.Message = opts.Message;
    }
}
export class DependencyFailedException extends __BaseException {
    constructor(opts) {
        super({
            name: "DependencyFailedException",
            $fault: "client",
            ...opts,
        });
        this.name = "DependencyFailedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, DependencyFailedException.prototype);
        this.Message = opts.Message;
    }
}
export class LoopDetectedException extends __BaseException {
    constructor(opts) {
        super({
            name: "LoopDetectedException",
            $fault: "server",
            ...opts,
        });
        this.name = "LoopDetectedException";
        this.$fault = "server";
        Object.setPrototypeOf(this, LoopDetectedException.prototype);
        this.Message = opts.Message;
    }
}
export class NotAcceptableException extends __BaseException {
    constructor(opts) {
        super({
            name: "NotAcceptableException",
            $fault: "client",
            ...opts,
        });
        this.name = "NotAcceptableException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NotAcceptableException.prototype);
    }
}
export const DialogState = {
    CONFIRM_INTENT: "ConfirmIntent",
    ELICIT_INTENT: "ElicitIntent",
    ELICIT_SLOT: "ElicitSlot",
    FAILED: "Failed",
    FULFILLED: "Fulfilled",
    READY_FOR_FULFILLMENT: "ReadyForFulfillment",
};
export class RequestTimeoutException extends __BaseException {
    constructor(opts) {
        super({
            name: "RequestTimeoutException",
            $fault: "client",
            ...opts,
        });
        this.name = "RequestTimeoutException";
        this.$fault = "client";
        Object.setPrototypeOf(this, RequestTimeoutException.prototype);
    }
}
export class UnsupportedMediaTypeException extends __BaseException {
    constructor(opts) {
        super({
            name: "UnsupportedMediaTypeException",
            $fault: "client",
            ...opts,
        });
        this.name = "UnsupportedMediaTypeException";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnsupportedMediaTypeException.prototype);
    }
}
export const ContentType = {
    GENERIC: "application/vnd.amazonaws.card.generic",
};
export const ActiveContextFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.parameters && { parameters: SENSITIVE_STRING }),
});
export const DialogActionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.slots && { slots: SENSITIVE_STRING }),
    ...(obj.message && { message: SENSITIVE_STRING }),
});
export const IntentSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.slots && { slots: SENSITIVE_STRING }),
});
export const GetSessionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.recentIntentSummaryView && {
        recentIntentSummaryView: obj.recentIntentSummaryView.map((item) => IntentSummaryFilterSensitiveLog(item)),
    }),
    ...(obj.sessionAttributes && { sessionAttributes: SENSITIVE_STRING }),
    ...(obj.dialogAction && { dialogAction: DialogActionFilterSensitiveLog(obj.dialogAction) }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
export const PostContentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.sessionAttributes && { sessionAttributes: SENSITIVE_STRING }),
    ...(obj.requestAttributes && { requestAttributes: SENSITIVE_STRING }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
export const PostContentResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.message && { message: SENSITIVE_STRING }),
    ...(obj.encodedMessage && { encodedMessage: SENSITIVE_STRING }),
    ...(obj.encodedInputTranscript && { encodedInputTranscript: SENSITIVE_STRING }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
export const PostTextRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.sessionAttributes && { sessionAttributes: SENSITIVE_STRING }),
    ...(obj.requestAttributes && { requestAttributes: SENSITIVE_STRING }),
    ...(obj.inputText && { inputText: SENSITIVE_STRING }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
export const PredictedIntentFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.slots && { slots: SENSITIVE_STRING }),
});
export const PostTextResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.alternativeIntents && {
        alternativeIntents: obj.alternativeIntents.map((item) => PredictedIntentFilterSensitiveLog(item)),
    }),
    ...(obj.slots && { slots: SENSITIVE_STRING }),
    ...(obj.sessionAttributes && { sessionAttributes: SENSITIVE_STRING }),
    ...(obj.message && { message: SENSITIVE_STRING }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
export const PutSessionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.sessionAttributes && { sessionAttributes: SENSITIVE_STRING }),
    ...(obj.dialogAction && { dialogAction: DialogActionFilterSensitiveLog(obj.dialogAction) }),
    ...(obj.recentIntentSummaryView && {
        recentIntentSummaryView: obj.recentIntentSummaryView.map((item) => IntentSummaryFilterSensitiveLog(item)),
    }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
export const PutSessionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.message && { message: SENSITIVE_STRING }),
    ...(obj.encodedMessage && { encodedMessage: SENSITIVE_STRING }),
    ...(obj.activeContexts && { activeContexts: SENSITIVE_STRING }),
});
